### PR TITLE
Backend/endpoints

### PR DESCRIPTION
**Pull request by pmh192**
Exposed endpoints related to team requests and getting teams in the context of tournaments.

- /api/tournament/{tournamentId}/team/all
- /api/tournament/{tournamentId}/team/complete
- /api/tournament/{tournamentId}/team/incomplete
- /api/user/requests/team
- /api/user/requests/pending
- /api/requests/{id}

### CODE REVIEW

File: backend/src/main/java/com/tourneynizer/tourneynizer/dao/TeamRequestDao.java [hunk 1]
```diff
@@ -56,31 +56,52 @@ public TeamRequest requestTeam(User user, Team team) {
...
-            preparedStatement.setLong(1, user.getId());
-            return preparedStatement;
-        }, idMapper);
+        return getRequestsHelper(idMapper, sql, user.getId());
+    }
+
+    public List<TeamRequest> getTeamRequests(Team team) {
+        String sql = "SELECT * FROM teamRequest WHERE team_id=? AND requester_id<>?;";
+        return getRequestsHelper(teamRequestRowMapper, sql, new Long[]{team.getId(), team.getCreatorId()});
+    }
```
Comment (lar-ryan-nick) [inline backend/src/main/java/com/tourneynizer/tourneynizer/dao/TeamRequestDao.java]: Why not also do a requests for team and by team?

File: backend/src/main/java/com/tourneynizer/tourneynizer/dao/TeamRequestDao.java [hunk 2]
```diff
@@ -104,12 +126,29 @@ public TeamRequest findById(long requestId) {
...
+        String sql = "UPDATE teamRequest SET accepted=False WHERE id=?";
+        int updated = this.jdbcTemplate.update(connection -> {
             PreparedStatement preparedStatement = connection.prepareStatement(sql);
             preparedStatement.setLong(1, teamRequest.getId());
             return preparedStatement;
         });
+
+        teamRequest.setAccepted(false);
+        return updated;
     }
```
Comment (lar-ryan-nick) [inline backend/src/main/java/com/tourneynizer/tourneynizer/dao/TeamRequestDao.java]: Are you still gonna be returning declined requests to the front end? Wouldn't declined requests be something that we wouldn't want to display so we should make declined requests hidden from the front end.

Comment (pmh192) [reply backend/src/main/java/com/tourneynizer/tourneynizer/dao/TeamRequestDao.java]: Not if the user wants to see that their request got declined.

File: backend/src/main/java/com/tourneynizer/tourneynizer/dao/TeamRequestDao.java (hunk 1 above)
Comment (pmh192) [reply backend/src/main/java/com/tourneynizer/tourneynizer/dao/TeamRequestDao.java]: That's for another PR. Currently there isn't an endpoint for a team to view its requests.

**Pull request merged**